controller/linodemachine_controller_helpers.go

```diff
@@ -46,6 +47,11 @@ import (
 // The decoded user_data must not exceed 16384 bytes per the Linode API
 const maxBootstrapDataBytes = 16384
 
+// Hard code the stack script ID for our CAPI shim stackscript.
+// This is a very specific stackscript which requires exact inputs
+// from this code, so it is unlikely we will want to configure it without a code change.
+const capiStackScriptID = 1319647
```

Comment:
I would guess this is hard-coded to simplify the user-experience (i.e. toggle a switch to use StackScript configurations). However, I think this decision raises some questions since there is now a unversion-controlled, externally-managed piece of code that has no direct link with CAPL versions. 

How do we handle updates (specifically breaking changes) to this specific StackScript? If we opaquely update this StackScript underneath CAPL users, new StackScript-based cluster installations for consumers using CAPL versions will break. If we decide to just update this reference every time we change the Metadata cloud-config, how do we enforce parity between the StackScript configuration method and the Metadata configuration method (especially during development)?

Reply:
Stackscripts have no concept of versioning. it is likely if we had a breaking interface change to a stackscript with relation to a CAPL change we would have to create a brand new stack script and point the new version of CAPL to that new stackscript